### DISCUSSION TITLE
Using `jj split` to create a merge commit

### DISCUSSION THREAD

**Discussion by FliegendeWurst** [jj-vcs/jj#5151] [Q&A] [answered]
I would like to do this transformation using jj split:

```
A      A'
|      |
|      M-----+
|      |     |
|      |     C
|      |     |
|      +-----+
|      |
B      B
```

As in, split A into two commits A' and C and make the commit graph look as above.

The reason for this is that I would then want to rebase C on a remote branch and push the result to open a PR. Currently I duplicate the commit and rebase the copy.
(If important: B is not equivalent to a remote branch. It roughly represents two merged remote branches.)

**Comment by jennings** [comment, marked as answer]
I would do this in two steps. First, split A into A' and C:

```shell
$ jj split
$ jj log
○  wxmwxkym cf27d17a
│  A
│
@  okmlklnu 56845831
│  C
│
○  urwtlokv cb7625b3
│  B
```

Next, create the merge commit:

```shell
# I'm using description() here to make it easier to see
# which commit is which
$ jj new -B description(A) -A description(C) -A description(B) -m M
$ jj log
○  wxmwxkym 16129332
│  A
│
@    owxwrlll f530545a
├─╮  (empty) M
│ │
○ │  okmlklnu 56845831
├─╯  C
│
○  urwtlokv cb7625b3
│  B

**Reply by FliegendeWurst** [reply]
That works, thank you. Didn't realize that `jj new` is so versatile!

**Comment by martinvonz** [comment]
What's the point of commit `M`?

**Reply by FliegendeWurst** [reply]
It's just the merge commit of B and C. Coming from git, I usually don't put changes in merge commits.

**Reply by martinvonz** [reply]
I was more wondering why it has C and C's parent (i.e. B) as ancestor. Why not have only C as parent? Is it that you want to model a no-ff merge?

**Reply by FliegendeWurst** [reply]
The reason is that I then do: `jj rebase -s description(C) --destination master@origin`, to open a PR against master with C.

So yes, I suppose that would be a no-ff merge.
